src/libraries/System.Private.CoreLib/src/System/Random.Xoshiro256StarStarImpl.cs

```diff
@@ -90,46 +95,42 @@ public override int Next()
 
             public override int Next(int maxValue)
             {
-                if (maxValue > 1)
+                ulong randomProduct = (ulong)maxValue * NextUInt32();
```

Comment:
I assume we are duplicating all this code from the other impl because the perf is significantly faster than calling through something. (This isn't new in this PR)

Reply:
I am not experienced at performance. But yes, I found a slight performance drop when having it call a common method. So I kind of assumed it was supposed to be duplicated.